### PR TITLE
Respect route targetPort in dynamic config

### LINKED ISSUE #6: Dynamic config manager ignores route targetPort
Consider the following scenario with router 3.11:

1. a service defines multiple ports (in the example below: `8080` and `8443`)
1. a route for that service sets `targetPort` so one of the service ports is preferred (in the example below: `8443`)
1. dynamic configuration manager is enabled

The initial HAProxy configuration after a full configuration reload is correct: it only includes the one port identified by `targetPort`. This can be verified using the HAProxy admin socket (starting with a single pod backing the service):

```
> show servers state be_tcp:test-haproxy-router:passthrough
# be_id be_name srv_id srv_name srv_addr srv_op_state srv_admin_state srv_uweight srv_iweight srv_time_since_last_change srv_check_status srv_check_result srv_check_health srv_check_state srv_agent_state bk_f_forced_id srv_f_forced_id srv_fqdn srv_port
44 be_tcp:test-haproxy-router:passthrough 1 pod:server-ssl-1-5m5n8:server-ssl:10.76.32.172:8443 10.76.32.172 2 0 256 256 27 6 3 4 6 0 0 0 - 8443
44 be_tcp:test-haproxy-router:passthrough 2 _dynamic-pod-1 172.4.0.4 0 5 1 1 27 1 0 0 14 0 0 0 - 8765
44 be_tcp:test-haproxy-router:passthrough 3 _dynamic-pod-2 172.4.0.4 0 5 1 1 27 1 0 0 14 0 0 0 - 8765
44 be_tcp:test-haproxy-router:passthrough 4 _dynamic-pod-3 172.4.0.4 0 5 1 1 27 1 0 0 14 0 0 0 - 8765
44 be_tcp:test-haproxy-router:passthrough 5 _dynamic-pod-4 172.4.0.4 0 5 1 1 27 1 0 0 14 0 0 0 - 8765
44 be_tcp:test-haproxy-router:passthrough 6 _dynamic-pod-5 172.4.0.4 0 5 1 1 27 1 0 0 14 0 0 0 - 8765
```

Let's now scale the deploymentConfig backing the multi-port service to 2 pods.

## Expected behavior

The dynamic list of HAProxy backend servers is updated with a single new endpoint pointing to the new pod and the route's `targetPort`

## Actual behaviour

The dynamic list of HAProxy backend servers is updated with servers for each port of the service, ignoring `targetPort`

```
> show servers state be_tcp:test-haproxy-router:passthrough
# be_id be_name srv_id srv_name srv_addr srv_op_state srv_admin_state srv_uweight srv_iweight srv_time_since_last_change srv_check_status srv_check_result srv_check_health srv_check_state srv_agent_state bk_f_forced_id srv_f_forced_id srv_fqdn srv_port
44 be_tcp:test-haproxy-router:passthrough 1 pod:server-ssl-1-5m5n8:server-ssl:10.76.32.172:8443 10.76.32.172 2 0 256 256 199 6 3 4 6 0 0 0 - 8443
44 be_tcp:test-haproxy-router:passthrough 2 _dynamic-pod-1 10.76.19.57 0 4 1 1 2 8 2 0 6 0 0 0 - 8443
44 be_tcp:test-haproxy-router:passthrough 3 _dynamic-pod-2 10.76.19.57 0 4 1 1 2 8 2 0 6 0 0 0 - 8080
44 be_tcp:test-haproxy-router:passthrough 4 _dynamic-pod-3 172.4.0.4 0 5 1 1 199 1 0 0 14 0 0 0 - 8765
44 be_tcp:test-haproxy-router:passthrough 5 _dynamic-pod-4 172.4.0.4 0 5 1 1 199 1 0 0 14 0 0 0 - 8765
44 be_tcp:test-haproxy-router:passthrough 6 _dynamic-pod-5 172.4.0.4 0 5 1 1 199 1 0 0 14 0 0 0 - 8765
```

It looks like this is due to the dynamic router using directly `service.EndpointTable` [here](https://github.com/openshift/router/blob/9ba3ab8/pkg/router/template/router.go#L619) and [here](https://github.com/openshift/router/blob/9ba3ab8/pkg/router/template/router.go#L666) instead of filtering ports with the template's [`endpointsForAlias` method](https://github.com/openshift/router/blob/a795299125925fb2156999d316f4438785c0a4a4/pkg/router/template/template_helper.go#L139)

I am testing a fix, will submit a PR shortly

### PR DESCRIPTION
**Pull request by alexcern**
Fix #6 

/cc @ramr

### CODE REVIEW

File: pkg/router/template/router.go [hunk 1]
```diff
@@ -616,8 +616,8 @@ func (r *templateRouter) dynamicallyAddRoute(backendKey string, route *routev1.R
 	newWeights := r.calculateServiceWeights(backend.ServiceUnits)
 	for key := range backend.ServiceUnits {
 		if service, ok := r.findMatchingServiceUnit(key); ok {
-			newEndpoints := service.EndpointTable
-			glog.V(4).Infof("For new route backend %s, replacing endpoints for service %s: %+v", backendKey, key, service.EndpointTable)
+			newEndpoints := endpointsForAlias(*backend, service)
```
Comment (alexcern) [inline pkg/router/template/router.go]: Not to reviewer: `endpointsForAlias` is defined in template_helper.go and intended to be called from the haproxy configuration template. It does exactly the job, but I'm not sure how fine it is to call it from outside the context of processing the template.

Comment (ramr) [reply pkg/router/template/router.go]: That's ok as its a helper function in the same package.

**Comment by ironcladlou** [comment]
Sorry it took so long to get to this.

/lgtm
/approve

**Pull request merged**